src/cmd/build.rs

```diff
@@ -352,6 +352,12 @@ fn do_optimization(
         .arg(format!("-O{}", optimization_level))
         .arg("-o")
         .arg(dest_optimized)
+
+        // the memory in our module is imported, `wasm-opt` needs to be told that
+        // the memory is initialized to zeros, otherwise it won't run the
+        // memory-packing pre-pass.
+        .arg("--zero-filled-memory")
```

Comment:
Does this also work with the old version of `wasm-opt`?

Reply:
That's a good point you raised. I updated the PR to show a helpful message in this case. That's how it currently looks:

```
ERROR: The wasm-opt optimization failed.               
A possible source of error could be that you have installed an outdated binaryen package.
                                                                               
The error which wasm-opt returned was:  
Unknown option '--zero-filled-memory' 
```

Reply:
Can we update the `README` to specify the minimum compatible installed version of `wasm-opt`? 

We could even check the version before executing to be at least `99` to be more accurate - if that is sufficient. But up to you.

Reply:
I added a note to the Readme!